pkg/loader/compose/compose.go

```diff
@@ -313,6 +476,125 @@ func (c *Compose) LoadFile(files []string) (kobject.KomposeObject, error) {
 	}
 
+	// Map the parsed struct to a struct we understand (kobject)
+	komposeObject, err := libComposeToKomposeMapping(composeObject)
+	if err != nil {
+		return kobject.KomposeObject{}, err
+	}
+
+	return komposeObject, nil
+}
+
+func dockerComposeToKomposeMapping(composeObject *types.Config) (kobject.KomposeObject, error) {
+
+	// Step 1. Initialize what's going to be returned
+	komposeObject := kobject.KomposeObject{
+		ServiceConfigs: make(map[string]kobject.ServiceConfig),
+		LoadedFrom:     "compose",
+	}
+
+	// Step 2. Parse through the object and conver it to kobject.KomposeObject!
+	// Here we "clean up" the service configuration so we return something that includes
+	// all relevant information as well as avoid the unsupported keys as well.
+	for _, composeServiceConfig := range composeObject.Services {
+
+		// Standard import
+		// No need to modify before importation
+		name := composeServiceConfig.Name
+		serviceConfig := kobject.ServiceConfig{}
+		serviceConfig.Image = composeServiceConfig.Image
+		serviceConfig.WorkingDir = composeServiceConfig.WorkingDir
+		serviceConfig.Annotations = map[string]string(composeServiceConfig.Labels)
+		serviceConfig.CapAdd = composeServiceConfig.CapAdd
+		serviceConfig.CapDrop = composeServiceConfig.CapDrop
+		serviceConfig.Expose = composeServiceConfig.Expose
+		serviceConfig.Privileged = composeServiceConfig.Privileged
+		serviceConfig.Restart = composeServiceConfig.Restart
+		serviceConfig.User = composeServiceConfig.User
+		serviceConfig.Stdin = composeServiceConfig.StdinOpen
+		serviceConfig.Tty = composeServiceConfig.Tty
+		serviceConfig.TmpFs = composeServiceConfig.Tmpfs
+		serviceConfig.ContainerName = composeServiceConfig.ContainerName
+		serviceConfig.Command = composeServiceConfig.Entrypoint
+		serviceConfig.Args = composeServiceConfig.Command
+
+		// This is a bit messy since we use yaml.MemStringorInt
+		// TODO: Refactor yaml.MemStringorInt in kobject.go to int64
+		// Since Deploy.Resources.Limits does not initialize, we must check type Resources before continuing
+		if (composeServiceConfig.Deploy.Resources != types.Resources{}) {
+			serviceConfig.MemLimit = libcomposeyaml.MemStringorInt(composeServiceConfig.Deploy.Resources.Limits.MemoryBytes)
+		}
+
+		// POOF. volumes_From is gone in v3. docker/cli will error out of volumes_from is added in v3
+		// serviceConfig.VolumesFrom = composeServiceConfig.VolumesFrom
+
+		// TODO
+		// Build is not yet supported, see:
+		// https://github.com/docker/cli/blob/master/cli/compose/types/types.go#L9
+		// We will have to *manually* add this / parse.
+		// serviceConfig.Build = composeServiceConfig.Build.Context
+		// serviceConfig.Dockerfile = composeServiceConfig.Build.Dockerfile
+
+		// Gather the environment values
+		// DockerCompose uses map[string]*string while we use []string
+		// So let's convert that using this hack
+		var envKeys []string
+		for key, _ := range composeServiceConfig.Environment {
```

Comment:
OHHHH. Woops. Change it to this:

```
		for key := range composeServiceConfig.Environment {
			envKeys = append(envKeys, key)
		}
```

Was a bit confused at first, I'll update it.